### DISCUSSION TITLE
0주차 Discussion

### DISCUSSION THREAD

**Discussion by chloe-codes1** [elegant-functional-2023/five-lines-of-code#1] [General]
스터디를 시작하기 전에 바라는 것, 얻어가고 싶은 것을 한마디씩 적어주세요 🤩

**Comment by hanana1253** [comment]
- 바라는 것
  - 겨울방학을 마치고 돌아온만큼 새로운 의지로 스터디 계속하기 (가늘고 길게~)
  - 매주 열심히 예습하기
  - 사과즙 배포
- 얻어가고 싶은 것
  - 화려한 코드 리팩토링 기술 
  - (하지만 이번에도 완독에 의의를...)

**Reply by sunhpark42** [reply]
- 사과즙 배포는 1번 했다.
- 완독 성공했다~

**Comment by ella-yschoi** [comment]
### 바라는 것

- 바보 같은 질문이라도 궁금하면 알아내서 내 것으로 만들기
- 현직 개발자 분들 사이에서 끝까지 살아남기 (?)
- 한결 님이 배포해 주시는 사과즙 감사히 먹기 🍎

### 얻어가고 싶은 것

- 면접에서 '이 스터디를 통해 무엇을 배웠나요?' 질문을 받으면 2가지 이상 답하기
- 책에 나온 내용을 기반으로 그동안의 프로젝트 리팩토링 + 블로그 포스팅 하기
- 스터디원 분들과 친해지기 ... ❣️

**Reply by sunhpark42** [reply]
- 하고싶은 질문, 이야기를 많이 했다.
- 사과즙 맛있었다!
- 이 스터디를 통해 무엇을 배웠나요? 적용을 해봐야 알 것 같다. 기존 작성한 코드의 리팩토링을 이 책 기반으로 할 것 같다. 해보면서 얻은거 블로깅 할 것 같아요.
- 친해진기는 회식을 기대하겠습니다.

**Comment by thgr8ganzi** [comment]
**바라는 것**

1. 포기하지 않고 책 완독하기

**얻어가고 싶은 것**

1. 머리에 조금이라도 남는 리팩토링 하기

**Reply by sunhpark42** [reply]
책 빠진 부분이 있었었다. 머리에 크게 남지 않은듯. 깊게 읽으려 했지만, 편하게 읽음. 다음 스터디는 더 열심히 해야겠다.

**Comment by bigsaigon333** [comment]
## 바라는 것
- 학습한 내용이 내 코드에 반영된다.
- 학습한 내용을 글로 잘 정리해서 축적하기

## 얻어가고 싶은 것
- 완독
- 우리팀에 들어오실분

**Reply by sunhpark42** [reply]
OOP 기반이 많아서 완전히는 아닌데 프로덕트에 조금씩 적용해보는중! 
글로 정리는 잘 안했다. 원래도 안함 ㅎㅎ.. 
중간에 빠지는 부분이 있었음. 다른 지인들은 뒷부분이 더 좋았다고 함. 나중에 다시 읽어보겠다.
ㅋㅋㅋㅋㅋ 우리팀 빨리 들어오세요~!

**Comment by chloe-codes1** [comment]
### 바라는 것

- 벼락치기 하지 않고 미리미리 책을 읽기
- 예제 코드를 따라 쳐보기

### 얻어 가고 싶은 것

- 리팩토링 패턴을 익혀 내 코드에 적용하기
- 팀에 적용할 만한 규칙을 정리해서 공유하기

**Reply by sunhpark42** [reply]
벼락치기로 책 읽었습니다.. 
예제코드 따라 쳐보지 않았다. ㅋㅋㅋㅋㅋㅋㅋ 오늘 회개 시간인가요
저번보다 나았던건 엥? 하고 넘어간 부분을 이번에는 엥? 스러운 부분을 파악을 잘 하고 넘어갔다.
- 팀에 적용할만한 새로운 것을 찾고자 했는데 공유할만한 내용은 찾지 못했다
- 책에서 얻은 지식 보다는, 책에서 언급된 키워드를 기반으로 각자의 경험을 나눌 수 있어 좋았다

**Comment by yunseorim1116** [comment]
**바라는 것**
사실 저는 이 책을 작년에 두 번정도 읽었는데요.. 그럼에도 불구하고 제 머릿속에 남은게 하나도 없었습니다. (자랑은 아니지만요.......)
이번에는 완독 목표 말고 정독으로 해서 남한테 말할 때 당당하게 "나 다 읽었다!" 라고 말할 수 있는 정도가 되고 싶어요. :<

**얻어 가고 싶은 것**
책의 좋은 부분은 적절히 가져가고 실무에 적용할 때 덜어낼 부분은 적당히 덜어낼 수 있는 선구안을 얻고 싶습니다~

**Reply by hanana1253** [reply]
두번이나 읽다니..!!!!!!!!!! 리스펙이네요..

**Reply by yunseorim1116** [reply]
따흐흑 반대로 말하면 두번이나 읽어도 이해가 안갓다는 소리 .................임니다.... (ⅈ▱ⅈ)
이번엔 정말 여러분과 함께 꼭꼭 씹어먹고싶어용~~~

**Reply by sunhpark42** [reply]
얻어가고 싶은 법 얻었습니다~!
좋은 부분은 리팩토링 하는 부분에서 내가 원하는 부분, 프론트에 적용할 부분에 적용해 보고 했어요! (완-전-멋-져)
문화적인 부분은 아직 적용할때가 아니라고 생각해서, 뒷부분이 오히려 별로였다.

나 다 읽었다! 는 소심하게 말할 수 있게 되었습니다.

**Comment by ddongule** [comment]
### 바라는 것
- 매 챕터 `다 & 잘` 읽어오기...^^ ... ... 

### 얻어 가고 싶은 것
- 나의 코드에 적용할 수 있는 리팩토링 기술 . . .

**Comment by sunhpark42** [comment]
바라는 것
- 책 완독
- 요즘 회사에서 코드 고칠일이 왕왕 있는데 더 나은 코드로 만들기
얻어가고 싶은 것
- 리팩토링 그게 뭔데..? 에 답하기

**Reply by sunhpark42** [reply]
완독은..했다.
코드 적용은 해보고 있다 (완전 잘은 아니지만)
그래도 문화적인 이야기 할 수 있었던게 좋았음.
리팩토링에 대한 생각은 매일 바뀌는 것 같다. 답변하기 어렵다.

**Comment by iborymagic** [comment]
바라는 것
- 책 제 때 읽기(급하게 읽지 말고..)

얻어가고 싶은 것
- 리팩토링 기술 하나라도 쓸만한거 얻어가기

**Comment by sa02045** [comment]
바라는 것
- 완독하기(sicp 때처럼만은...)

얻어가고 싶은 것
- 챕터 잘 읽고 나만의 코드 원칙 정립하기

**Reply by sunhpark42** [reply]
완독은 했다! SCIP는 뒷부분이 남아서 찝찝했는데, 완독했다는 성취감이 있었다.
코드 원칙을 정립하고 싶었는데 오히려 헷갈려졌다. 선택지가 왕왕 늘어난 느낌. 하나의 원칙을 세운다기보다는 선택지가 넓어져서, 아 이런 방법도 있었지 하고 책 다시 찾아볼 수 있을 것 같음.
